content/departments/engineering/dev/tools/grpc_cs_enablement.md

```diff
@@ -0,0 +1,110 @@
+# gRPC CS Enablement Guide
+
+# **What is gRPC?**
+
+[gRPC](https://grpc.io/) is a high-performance [RPC](https://en.wikipedia.org/wiki/Remote_procedure_call) framework developed by Google.
+
+## What is changing?
+
+In Sourcegraph `5.2.X`, we are enabling gRPC by default (with the option to disable it in case of an issue). gRPC will be the only option in the `5.3.X` release.
+
+## **Why are we making this change?**
```

Comment:
I'd also mention some practical benefits for our usecase: it's hard to follow references and find out exactly where an endpoint is implemented, we had multiple different representations and no standard way, when you added a new endpoint you had too many options to choose from, etc.

Reply:
I added a few more to the list:

> - Large 3rd party plugin ecosystem
> - Better discoverability and documentation: Finding [what RPCs are available](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@f28a7fab171cacefa66553d875069b8338ee7308/-/blob/internal/gitserver/v1/gitserver_grpc.pb.go?L42-59#tab=references), [where they are implemented](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@f28a7fab171cacefa66553d875069b8338ee7308/-/blob/internal/gitserver/v1/gitserver_grpc.pb.go?L328:2-328:6#tab=implementations_go), and [who is calling them](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@f28a7fab171cacefa66553d875069b8338ee7308/-/blob/internal/gitserver/v1/gitserver_grpc.pb.go?L46:2-46:6#tab=references) is as easy as using “find references”.